app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/SessionDataExtractor.scala

```diff
@@ -37,4 +41,45 @@ trait SessionDataExtractor extends Results {
         Future.successful(Redirect(routes.StartController.start()))
     }
 
+  def withAnswersAndRoutes[T](
+    f: (FillingOutClaim, Option[T], ReimbursementRoutes) => Future[Result]
+  )(implicit extractor: DraftC285Claim => Option[T], request: RequestWithSessionData[_]): Future[Result] =
+    request.sessionData.flatMap(_.journeyStatus) match {
+      case Some(fillingOutClaim @ FillingOutClaim(_, _, draftClaim: DraftClaim)) =>
+        val numOfClaims = getNumberOfClaims(draftClaim)
+        val refType     = getMovementReferenceNumber(draftClaim)
+        val router      = getRoutes(numOfClaims, refType)
+        draftClaim
+          .fold(extractor(_))
+          .fold[Future[Result]](f(fillingOutClaim, None, router))(data => f(fillingOutClaim, Option(data), router))
+      case _                                                                     =>
+        Future.successful(Redirect(baseRoutes.StartController.start()))
+    }
+
+  def getNumberOfClaims(draftClaim: DraftClaim): Option[SelectNumberOfClaimsType] =
+    draftClaim
+      .fold(identity)
+      .selectNumberOfClaimsAnswer
+      .flatMap(
+        _.fold(_.selectNumberOfClaimsChoice, a => Option(a.selectNumberOfClaimsChoice))
+      )
+
+  def getMovementReferenceNumber(draftClaim: DraftClaim): Option[MovementReferenceNumber] =
+    draftClaim.fold(identity).movementReferenceNumber
+
+  def getRoutes(
+    numberOfClaims: Option[SelectNumberOfClaimsType],
+    mrnOrEntryNmber: Option[MovementReferenceNumber]
```

Comment:
Rather than `MovementReferenceNumber` it might be better if you had a `MovementReferenceNumberType`. Then you wouldn't need to call it with a synthetic MRN.